admin/class-acc_user_importer-admin.php

```diff
@@ -358,148 +374,192 @@ private function proccess_user_data ( $users ) {
+					$role_refreshed[] = $user_contact_id;
+					$role_refreshed_email[] = $user_email;
+				}
+
+				//Check if email changed
+				if (!$update_this_user) {
+					if ($user_email != $user_meta->user_email) {
+						$this->log_dual(" > email changed from " . $user_meta->user_email . " to " . $user_email);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if HomePhone changed
+				if (!$update_this_user) {
+					if ($user["HomePhone"] != $user_meta->home_phone) {
+						$this->log_dual(" > home phone changed from " . $user_meta->home_phone . " to " . $user["HomePhone"]);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if Cell phone changed
+				if (!$update_this_user) {
+					if ($user["Cell Phone"] != $user_meta->cell_phone) {
+						$this->log_dual(" > cell phone changed from " . $user_meta->cell_phone . " to " . $user["Cell Phone"]);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if MEMBERSHIP_N changed
+				if (!$update_this_user) {
+					if ($user["MEMBERSHIP_N"] != $user_meta->membership) {
+						$this->log_dual(" > membership# changed from " . $user_meta->membership . " to " . $user["MEMBERSHIP_N"]);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if Membership Expiry changed
+				if (!$update_this_user) {
+					if ($user["Membership Expiry Date"] != $user_meta->expiry) {
+						$this->log_dual(" > expiry changed from " . $user_meta->expiry . " to " . $user["Membership Expiry Date"]);
+						$update_this_user = true;
+					}
+				}
+
+				//Check if city changed
+				if (!$update_this_user) {
+					if ($user["City"] != $user_meta->city) {
+						$this->log_dual(" > city changed from " . $user_meta->city . " to " . $user["City"]);
+						$update_this_user = true;
+					}
 				}
-			
-				//add user meta if update succeded
-				else {
-					
-					//FLAG: editions par kfg en juin 2020 => Ajouté les conditions
-					if(!empty($user["HomePhone"]))
-						update_user_meta( $user_id, 'home_phone', $user['HomePhone'] );
-					if(!empty($user["Cell Phone"]))
-						update_user_meta( $user_id, 'cell_phone', $user['Cell Phone'] );
-					if(!empty($user["MEMBERSHIP_N"]))
-						update_user_meta( $user_id, 'membership', $user['MEMBERSHIP_N'] );
-					if(!empty($user["Membership Expiry Date"]))
-						update_user_meta( $user_id, 'expiry', $user['Membership Expiry Date'] );
-					if(!empty($user["City"]))
-						update_user_meta( $user_id, 'city', $user['City'] );
 
+				if (!$update_this_user) {
+					$this->log_dual(" > Nothing changed for this user");
+				}
+			}
+
+			//only update user if needed
+			if ($update_this_user) {
+				//update core info (name, email)
+				$wp_user = wp_insert_user( $user_data );
+				if ( is_wp_error($wp_user) ) {
+					$this->log_dual(" > failed to update user");
+					$this->log_dual(" > WP:" . $wp_user->get_error_message());
+
+				} else {
+					//update user meta
+					update_user_meta( $wp_user, 'home_phone', $user['HomePhone'] );
```

Comment:
Again, this will produce notices if the `HomePhone` (or any other index) is missing.

Reply:
I will let you do the changes.